kernel/redFlags.ml

```diff
@@ -59,7 +59,10 @@ let red_add red = function
     { red with r_const = { r with tr_cst = Cpred.add kn r.tr_cst } }
   | PROJ p ->
     let r = red.r_const in
-    { red with r_const = { r with tr_prj = PRpred.add p r.tr_prj } }
+    { red with r_const =
+                 { r with tr_prj = PRpred.add p r.tr_prj;
```

Comment:
This looks like a regression to me. We want to make the kernel as agnostic as possible towards compat constants, not hardwire more behaviours relying on the proj-constant binding.